### PR TITLE
Maven target clean ups and test about consideration of changed BND-instructions

### PR DESCRIPTION
**Pull request by HannesWell**
This PR adds the test-case initially created for https://github.com/eclipse-m2e/m2e-core/pull/1386 about consideration of changed BND-instructions.
Furthermore it cleans up the code added in https://github.com/eclipse-m2e/m2e-core/pull/1387 a bit and fixes two deprection warnings.

### CODE REVIEW

File: org.eclipse.m2e.pde.target/src/org/eclipse/m2e/pde/target/shared/WrappedBundle.java [hunk 1]
```diff
@@ -14,72 +14,22 @@
...
 	/**
 	 * @return the messages that where produced
 	 */
 	public Stream<ProcessingMessage> messages() {
-		return Stream.concat(messages.stream(), depends.stream().flatMap(dep -> dep.messages()));
-	}
-
-	@Override
-	public int hashCode() {
-		return Objects.hash(instructionsKey, node);
```
Comment (HannesWell) [inline org.eclipse.m2e.pde.target/src/org/eclipse/m2e/pde/target/shared/WrappedBundle.java:67]: As far as I can tell this class (now record) is not used as key in any map.
Therefore I think it is not necessary to override hashCode/equals.

Comment (laeubi) [reply org.eclipse.m2e.pde.target/src/org/eclipse/m2e/pde/target/shared/WrappedBundle.java:67]: The idea was/is that code can use this to decide if two items are equal without having to know the internal state, even though it is currently not used (at m2e) in such a way.

File: org.eclipse.m2e.pde.target/src/org/eclipse/m2e/pde/target/shared/WrappedBundle.java [hunk 2]
```diff
@@ -14,72 +14,22 @@
...
-		return jar;
-	}
-
-	/**
-	 * @return the file where the wrappes bundle is located
-	 */
-	public Path getFile() {
-		return file;
-	}
+public record WrappedBundle(DependencyNode node, List<WrappedBundle> depends, String instructionsKey,
```
Comment (laeubi) [inline org.eclipse.m2e.pde.target/src/org/eclipse/m2e/pde/target/shared/WrappedBundle.java:23]: This should not be a record as it is not a pure container, this makes all its members public to the outside world what is not desired.

File: org.eclipse.m2e.pde.target/src/org/eclipse/m2e/pde/target/shared/WrappedBundle.java (hunk 1 above)
Comment (HannesWell) [reply org.eclipse.m2e.pde.target/src/org/eclipse/m2e/pde/target/shared/WrappedBundle.java]: Acknowledged.

**Pull request merged**